gazelle/python/parser.go

```diff
@@ -164,9 +165,32 @@ func (p *python3Parser) parse(pyFilenames *treeset.Set) (*treeset.Set, map[strin
 				mainModules[res.FileName].Add(m)
 			}
 		}
+
+		// Collect all annotations from each file into a single annotations struct.
+		for k, v := range annotations.ignore {
+			allAnnotations.ignore[k] = v
+		}
+		allAnnotations.includeDep = append(allAnnotations.includeDep, annotations.includeDep...)
 	}
 
-	return modules, mainModules, nil
+	allAnnotations.includeDep = removeDupesFromStringTreeSetSlice(allAnnotations.includeDep)
+
+	return modules, mainModules, allAnnotations, nil
+}
+
+// removeDupesFromStringTreeSetSlice takes a []string, makes a set out of the
+// elements, and then returns a new []string with all duplicates removed. Order
+// is preserved.
+func removeDupesFromStringTreeSetSlice(array []string) []string {
+	s := treeset.NewWith(godsutils.StringComparator)
+	for _, v := range array {
+		s.Add(v)
+	}
+	dedupe := make([]string, s.Size())
+	for i, v := range s.Values() {
+		dedupe[i] = fmt.Sprint(v)
+	}
+	return dedupe
```

Comment:
Coming from python, this function feels icky to me. I'd really like to be able to just do

```python
allAnnotations.includeDep = list(set(allAnnotations.includeDep))
```

above but I couldn't figure out how to do so. The issue is related to types: the `treeset.Values()` is type `[]interface{}`, `treeset.NewWith`'s `values` arg is `interface{}`, but `includeDep` is `[]string`. From what I could find, there's no easy way to "cast" `[]string` to `[]interface{}` and back.

If you have any suggestions on how to clean things up, I'd be happy to hear them. Otherwise no reply is needed.

Reply:
I've been meaning to pick up https://github.com/f0rmiga/datanalgo and add the missing data structures using generics. I can try to resurrect that, which would be ideal to use with the Gazelle extension.